plugins/index.js

```diff
@@ -90,6 +92,7 @@ export function loadPlugin(manifest) {
 
         if (oldManifest) {
             // upgrading, perform cleanup
+            hideRHSPlugin(oldManifest.id);
```

Comment:
It's slightly annoying to have an action creator to dispatch an event that could conceivably just be handled by the `REMOVED_WEBAPP_PLUGIN` within the reducers. Would it be possible to store the real `pluginId` alongside the `pluginComponentId` in the store and just key off the existing event below?

Reply:
Even if I were to store `pluginId` and `pluggableId` in the store (as reducers) I will still not be able to close RHS from within the reducer as  the [`rhsState`](https://github.com/mattermost/mattermost-webapp/blob/b806d43982e4d34beca05aa632f3f9cc3263cacd/reducers/views/rhs.js#L93) reducer does not have any knowledge about which `plugginId` or `pluggableId` is currently set. Which means I'll have to either:

1. Share state between reducers
   * Not recommended as it'll cause more confusion down the line

2. Include more data when dispatching `REMOVED_WEBAPP_PLUGIN` that can help `rhsState` make the proper deicion
3.

Reply:
This approach is fine for now. I think the problem with `reducers/views/rhs.js` is that everything is so granular. If we merged some of the reducers together and stored an object for each, it would be easy to key off `REMOVED_WEBAPP_PLUGIN` in the right conditions with presumably no other changes to the selectors and such.

Reply:
That's true. Combining them will work. Just need to put some thought into which ones to combine.